gnomad_qc/v4/annotations/generate_freq.py

```diff
@@ -1017,7 +1048,24 @@ def main(args):
                 "Splitting VDS by ukb_sample annotation to reduce data size for"
                 " densification..."
             )
-            vds_dict = split_vds_by_strata(vds, strata_expr=vds.variant_data.ukb_sample)
+            # For v4.0 the line of code this comment block was:
+            # vds_dict = split_vds_by_strata(vds, strata_expr=vds.variant_data.ukb_sample)
+            #
+            # This uses hail's hl.vds.filter_samples() function and keeps the default
+            # remove_dead_alleles=False. However, due to unexpected behavior in the
+            # code, the hl.vds.filter_samples() function still removes dead alleles by
+            # removing any variant row that does not have a defined entry. When the
+            # VDS is densified, this results in a loss of all homozygous reference calls
+            # in the reference data sparse MT at these dead allele sites. Because we
+            # had to split the VDS for v4.0 with intentions of rejoining the two subset
+            # VDSs across all variant sites, this unexpected behavior resulted in the
+            # loss of all homozygous reference calls in the reference data sparse MT and
+            # thus reduced Allele Number (AN) for any bi-allelic variant exclusive to
+            # one of the subsetted VDSs. The replacement code below uses our own custom
+            # function to split the VDS and keep the dead alleles in the reference data.
+            # It was never run in production, but it is the correct way to split the VDS
```

Comment:
So we didn't rerun this but we have separate production codes to correct freq and to get all sites AN? Do you think it's worth noting the fix here? I'm looking at this PR https://github.com/broadinstitute/gnomad_qc/pull/556 for fix_freq_an.py.

Reply:
Yeah good point, I'll reference the script and ticket.